src/network/wait-online/manager.c

```diff
@@ -352,6 +354,13 @@ static int on_dns_configuration_event(
         assert(link);
 
         if (error_id) {
+                if (streq(error_id, SD_VARLINK_ERROR_DISCONNECTED)) {
+                        log_debug("DNS configuration monitor disconnected, reconnecting...");
+
+                        m->varlink_client = sd_varlink_unref(m->varlink_client);
```

Comment:
i usually like properly encapsulated calls, i.e. i'd move this line into manager_dns_configuration_listen(), right before the new varlink client is installed. after all, unreffing NULL is a NOP, and hence calling it if not necessary has no negative effect. That would make manager_dns_configuration_listen() very robust: it would do the right thing regardless of the context it is called in.

but doesn't really matter

Reply:
Makes sense, and fixed.